File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F06.P2.zig
const std = @import("std");

fn probe() i32 {
// BEGIN PROBE F06.P2
const q, const r = divmod2(7, 3);
return q + r;
}

fn divmod2(a: i32, b: i32) struct { i32, i32 } {
    return .{ @divTrunc(a, b), @rem(a, b) };
}
// END PROBE F06.P2

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F06.P2.rs
fn probe() -> i32 {
// BEGIN PROBE F06.P2
fn divmod2(a: i32, b: i32) -> (i32, i32) { (a / b, a % b) }
let (q, r) = divmod2(7, 3);
q + r
// END PROBE F06.P2
}

fn main() {
    println!("{}", probe());
}
